photutils/background/core.py

```diff
@@ -0,0 +1,675 @@
+                        unicode_literals)
+from distutils.version import LooseVersion
+import abc
+import numpy as np
+import warnings
+from astropy.extern import six
+from astropy.utils.misc import InheritDocstrings
+from astropy.utils.exceptions import AstropyUserWarning
+from astropy.stats import (sigma_clip, mad_std, biweight_location,
+                           biweight_midvariance)
+
+import astropy
+if LooseVersion(astropy.__version__) < LooseVersion('1.1'):
+    ASTROPY_LT_1P1 = True
+else:
+    ASTROPY_LT_1P1 = False
+
+
+__all__ = ['BkgBase', 'BackgroundBase', 'BackgroundRMSBase',
+           'MeanBackground', 'MedianBackground', 'MMMBackground',
+           'SExtractorBackground', 'BiweightLocationBackground',
+           'StdBackgroundRMS', 'MADStdBackgroundRMS',
+           'BiweightMidvarianceBackgroundRMS']
+
+
+class _ABCMetaAndInheritDocstrings(InheritDocstrings, abc.ABCMeta):
+    pass
+
+
+@six.add_metaclass(_ABCMetaAndInheritDocstrings)
+class BkgBase(object):
+    """
+    Base class for Background and Background RMS classes.
+
+    Parameters
+    ----------
+    sigma : float, optional
+        The number of standard deviations to use for both the lower and
+        upper clipping limit. These limits are overridden by
+        ``sigma_lower`` and ``sigma_upper``, if input. Defaults to 3.
+    sigma_lower : float or `None`, optional
+        The number of standard deviations to use as the lower bound for
+        the clipping limit. If `None` then the value of ``sigma`` is
+        used. Defaults to `None`.
+    sigma_upper : float or `None`, optional
+        The number of standard deviations to use as the upper bound for
+        the clipping limit. If `None` then the value of ``sigma`` is
+        used. Defaults to `None`.
+    iters : int or `None`, optional
+        The number of iterations to perform sigma clipping, or `None` to
+        clip until convergence is achieved (i.e., continue until the
+        last iteration clips nothing). Defaults to 5.
+    """
+
+    def __init__(self, sigclip=True, sigma=3, sigma_lower=None,
+                 sigma_upper=None, iters=5):
+
+        self.sigclip = sigclip
+        self.sigma = sigma
+        self.sigma_lower = sigma_lower
+        self.sigma_upper = sigma_upper
+        self.iters = iters
+
+    def sigma_clip(self, data):
+
+        if ASTROPY_LT_1P1:
+            warnings.warn('sigma_lower and sigma_upper will be ignored '
+                          'because they are not supported astropy < 1.1',
+                          AstropyUserWarning)
+            return sigma_clip(data, sig=self.sigma,
+                              iters=self.iters)
+        else:
+            return sigma_clip(data, sigma=self.sigma,
+                              sigma_lower=self.sigma_lower,
+                              sigma_upper=self.sigma_upper,
+                              iters=self.iters)
+
+
+class BackgroundBase(BkgBase):
```

Comment:
I think `BackgroundBase` and `BkgBase` are a confusing pair.  I get the intended purpose, but I think it's a bit confusing as it stands now.  I'll leave a general comment about what I think might be a better herirarchy, but even if you don't like that, this name should change.  Maybe `BackgroundBaseValue` and `BackgroundBase`?  Or `BackgroundBaseCentral`?